src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1731 @@
+        visual_output = visual_output.reshape(visual_output.shape[0], visual_output.shape[1], -1)
+        # shape = [*, grid ** 2, hidden_size]
+        visual_output = visual_output.permute(0, 2, 1)
+        t = self.class_embedding.to(visual_output.dtype) + torch.zeros(
+            visual_output.shape[0], 1, visual_output.shape[-1], dtype=visual_output.dtype, device=visual_output.device
+        )
+        # shape = [*, grid ** 2 + 1, hidden_size]
+        visual_output = torch.cat([t, visual_output], dim=1)
+        visual_output = visual_output + self.positional_embedding.to(visual_output.dtype)
+        visual_output = self.ln_pre(visual_output)
+        # NLD -> LND
+        visual_output = visual_output.permute(1, 0, 2)
+
+        visual_outputs = self.transformer(visual_output, attention_mask)
+        # shape = [num_hidden_layers, hidden_size, *, grid ** 2]
+        visual_outputs = torch.stack(visual_outputs, dim=0)
+        # shape = [num_hidden_layers, *, hidden_size, grid ** 2]
+        visual_outputs = visual_outputs.permute(0, 2, 1, 3)
+        if self.share_layernorm:
+            visual_outputs = self.ln_post(visual_outputs)
+        else:
+            visual_outputs_stack = []
+            for visual_output, ln in zip(visual_outputs, self.ln_separate):
+                visual_output = ln(visual_output)
+                visual_outputs_stack.append(visual_output)
+            # shape = [num_hidden_layers, *, hidden_size, grid ** 2]
+            visual_outputs = torch.stack(visual_outputs_stack, dim=0)
+        return visual_outputs
+
+    def forward_pre(self, hidden_state: torch.Tensor):
+        # shape = [*, hidden_size, grid, grid]
+        visual_outputs_pre = self.conv1(hidden_state)
+        # shape = [*, hidden_size, grid ** 2]
+        visual_outputs_pre = visual_outputs_pre.reshape(visual_outputs_pre.shape[0], visual_outputs_pre.shape[1], -1)
+        # shape = [*, grid ** 2, hidden_size]
+        visual_outputs_pre = visual_outputs_pre.permute(0, 2, 1)
+        embeddings_to = self.class_embedding.to(visual_outputs_pre.dtype) + torch.zeros(
+            visual_outputs_pre.shape[0],
+            1,
+            visual_outputs_pre.shape[-1],
+            dtype=visual_outputs_pre.dtype,
+            device=visual_outputs_pre.device,
+        )
+        # shape = [*, grid ** 2 + 1, hidden_size]
+        visual_outputs_pre = torch.cat([embeddings_to, visual_outputs_pre], dim=1)
+        visual_outputs_pre = visual_outputs_pre + self.positional_embedding.to(visual_outputs_pre.dtype)
+        visual_outputs_pre = self.ln_pre(visual_outputs_pre)
+        # NLD -> LND
+        visual_outputs_pre = visual_outputs_pre.permute(1, 0, 2)
+        return visual_outputs_pre
+
+    def forward_post(self, hidden_state: torch.Tensor):
+        visual_output_post = hidden_state.permute(1, 0, 2)
+        visual_output_post = self.ln_post(visual_output_post)
+        return visual_output_post
+
+
+class LinkTower(nn.Module):
```

Comment:
This should be prefixed with `BridgeTower`.

Reply:
We have fixed this in our latest commit. Thanks for the feedback!